src/peft/utils/other.py

```diff
@@ -496,3 +496,39 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
         unique_id = storage_ptr(tensor)
 
     return tensor.device, unique_id, storage_size(tensor)
+
+
+# adapted from https://github.com/Bavest/fin-llama/blob/8ba68f4b49753358ffbe693d38f2461c031625e3/qlora.py#L229
+def get_linear_layer_names(model):
```

Comment:
Given that we have #1295, do you think we still need this option? At the very least, we could try to re-use most of the code between this function and that PR. For instance, this function does not consider bnb layers.

Reply:
Given that bnb layers inherit from Linear layers, this function is enough to get the intended functionality. Will take a look at the other PR.

Reply:
Didn't now about bnb layers inheriting from Linear! Logic in #1295 was initially based on the special handling in the Qlora repo, but it's simplified now!